docs/plasmapy_sphinx/extras.py

```diff
@@ -0,0 +1,109 @@
+"""
+This module contains extra (optional) sphinx setup parameters that are not need
+to run the `plasmapy_sphinx` extension but are useful for documenting.  See
+`plasmapy_sphinx.extras.setup` for further details.
+"""
+
+import re
+
+from sphinx import addnodes
+from sphinx.application import Sphinx
+from sphinx.domains.python import PyField
+from sphinx.locale import _
+from sphinx.util.docfields import Field, GroupedField
+
+
+def parse_event(env, sig, signode):
+    """
+    Used to setup the ``event`` directive and role for documenting Sphinx events.
+    Taken from the ``conf.py`` file of `Sphinx's documentation
+    <https://github.com/sphinx-doc/sphinx/blob/
+    8653ceca0021f6ac6ff0aac6c26e2a455c6d4b21/doc/conf.py#L123-L138>`_.
+
+    Parameters
+    ----------
+    env : sphinx.environment.BuildEnvironment
+        Instance of the Sphinx's build environment.
+
+    sig : str
+        The "signature" given the the event directive or role.  For example,
+
+        .. code-block:: rst
+
+            .. event:: foo(bar)
+
+            :event:`foo`
+
+        in the directive case ``foo(bar)`` would be the signature and in the role
+        case ``foo`` would be the signature.
+
+    signode : sphinx.addnodes.desc_signature
+        A `docutils` Node for the object signatures.
+    """
+    event_sig_re = re.compile(r"([a-zA-Z-]+)\s*\((.*)\)")
+
+    m = event_sig_re.match(sig)
+    if not m:
+        signode += addnodes.desc_name(sig, sig)
+        return sig
+    name, args = m.groups()
```

Comment:
If this were Python 3.8+ only, then this would have been a good place for a walrus `:=` operator.  Alas!  

[optional] I'm wondering if there would be a more descriptive variable name than `m`.  Not high priority, though.

Reply:
This was copied and pasted straight from sphinx's `conf.py`.  That's why it does not resemble my coding style.